pkg/visitors/createfv.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2018 the LinuxBoot Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package visitors
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"strconv"
+	"unsafe"
+
+	"github.com/linuxboot/fiano/pkg/uefi"
+)
+
+// CreateFV creates a firmware volume at given offset
+type CreateFV struct {
+	AbsOffset uint64
+	Size      uint64
+	offset    uint64
+	found     bool
+}
+
+// Run wraps Visit and performs some setup and teardown tasks.
+func (v *CreateFV) Run(f uefi.Firmware) error {
+	err := f.Apply(v)
+	if err != nil {
+		return err
+	}
+	if !v.found {
+		return fmt.Errorf("Cannot create FV at %#x (+%#x) no BIOS region found", v.AbsOffset, v.Size)
+	}
+	return nil
+}
+
+// Visit applies the Remove visitor to any Firmware type.
+func (v *CreateFV) Visit(f uefi.Firmware) error {
+	if v.found {
+		return nil
+	}
+	var offset, end uint64
+	switch f := f.(type) {
+	case *uefi.BIOSRegion:
+		if f.FRegion != nil {
+			offset = uint64(f.FRegion.BaseOffset())
+			end = uint64(f.FRegion.EndOffset())
+		}
+		fmt.Printf("BIOS     %#08x-%#08x\n", offset, end)
```

Comment:
I assume at some point this becomes a debug print?

Reply:
I'm going to drop it entierly